### PR TITLE
Overhaul swingset configuration

### PR DESCRIPTION
**Pull request by FUDCo**
These changes implement the new swingset configuration model per #1331.  A writeup of the deltas is in the file `APIChange.md` (which should probably be rehomed or renamed or incorporated into some other file, but I wanted to get these changes documented without delaying for the time it would take me to get clear on the change notes publication flow).

### CODE REVIEW

**Comment by warner** [comment]
It looks like some more changes landed on trunk after you branched off (but not mine, so I think they won't conflict). I'm going to experiment with github's "Update branch" button and see what it does. I think we'll want to rebase this whole thing before landing, to avoid the little rats nest of branches that this button will probably create.

**Comment by warner** [comment]
Yep, that button created a merge commit between the branch you pushed and current trunk, which is more messy than we'd want to land. But a `git rebase master` cleans it up just fine, so let's remember to either do that manually, or use the Merge button with the "Rebase" option.

File: packages/SwingSet/src/kernel/kernel.js
Comment (warner) [inline packages/SwingSet/src/kernel/kernel.js]: I don't think this should take `vatOptions`: if those are the parameters that will get passed into `buildRootObject`'s second argument, then they have to be supplied to the function that creates the manager originally. `addVatManager` takes a manager that's already been built and wires it into the delivery/syscall pathways, but by that point it's too late to provide anything to `buildRootObject`.

(also, it doesn't look like `vatOptions` are used within this function, which should have triggered a lint error)

`creationOptions`, yes.

File: packages/SwingSet/src/kernel/kernel.js
Comment (warner) [inline packages/SwingSet/src/kernel/kernel.js:689]: eventually these bundles will want to move into kernelDB, or maybe a separate store of some sort, but definitely on-disk rather than in-memory

File: packages/SwingSet/src/kernel/kernel.js
Comment (warner) [inline packages/SwingSet/src/kernel/kernel.js]: let's add a check here to throw if the bundleName is not in the table, I think that's an easy error to trigger from userspace so we should give a clear diagnostic, rather than having `createVatDynamically` emit something obscure when `importBundle` tries to deal with `undefined`

File: packages/SwingSet/test/test-transcript.js
Comment (warner) [inline packages/SwingSet/test/test-transcript.js]: should this `cfg.hostStorage = s` be removed, replaced by the `{ hostStorage: s}` done below?

**Review by warner — CHANGES_REQUESTED** [review]
Looks pretty good. I"m still nervous about YAGNI, but I like everything here. A few minor changes to make, feel free to land once those are done.

File: packages/SwingSet/src/controller.js
Comment (FUDCo) [reply packages/SwingSet/src/controller.js]: I think the language in this comment was a hangover from an earlier iteration of the config design.  Will update.

File: packages/SwingSet/src/controller.js
Comment (FUDCo) [reply packages/SwingSet/src/controller.js]: I'm thinking that `parameters` may be better, simply because we tend to use `options` or `blahblahblahOptions` as a final parameter in a function's args, and although that's what this translates to internally (hence my thinking when I used the word "options" here), from the perspective of somebody authoring a config file or composing a config object that's not how they use it.  If we want to make this change I'm actually inclined to do it now, since it just revises a bunch of files that are already touched in this PR to touch them slightly differently; it won't affect the size of the PR itself or the volume of stuff to be reviewed significantly if at all.

File: packages/SwingSet/src/controller.js
Comment (FUDCo) [reply packages/SwingSet/src/controller.js:361]: Yes.  I thought about letting a bundle reference another bundle (which seems at least logically consistent) and I couldn't think of use case for it that would justify all the jumping through hoops that would be needed to make it actually work.

File: packages/SwingSet/src/kernel/kernel.js
Comment (FUDCo) [reply packages/SwingSet/src/kernel/kernel.js]: I think that's right.  I suspect this occurrence of `throw new Error` resulted from copy and pasting from some similar code this was derived from.  BTW, when I went to clean this stuff up I discovered there's a ton of `throw new Error`s in `kernel.js`

File: packages/SwingSet/src/kernel/kernel.js
Comment (FUDCo) [reply packages/SwingSet/src/kernel/kernel.js]: I think you are right.  This may have slipped in during debugging.  There were some earlier intermediate states where lint was complaining, so I'm not sure what's going on with that here.  But I'll take it out.

This points out a coding issue worthy of a longer discussion elsewhere: I think the options bag pattern does not compose well.

File: packages/SwingSet/src/kernel/kernel.js
Comment (FUDCo) [reply packages/SwingSet/src/kernel/kernel.js:689]: Yes; this is a transitional solution.

File: packages/SwingSet/src/kernel/kernel.js
Comment (warner) [reply packages/SwingSet/src/kernel/kernel.js]: yeah, not worth cleaning up the old ones now, but as we visit code we can migrate everythng to plain `throw Error`

File: packages/SwingSet/src/kernel/kernel.js
Comment (FUDCo) [reply packages/SwingSet/src/kernel/kernel.js]: Good catch, er, so to speak.  Will do.

File: packages/SwingSet/test/test-transcript.js
Comment (FUDCo) [reply packages/SwingSet/test/test-transcript.js]: Yup.

**Pull request merged**
